pkg/syslog/rfc5424/parser.go

```diff
@@ -19,7 +19,7 @@ func Parse(buf []byte) (*Message, error) {
 	}
 
 	if cursor < len(buf) {
-		msg.Msg = buf[cursor:]
+		msg.Msg = buf[cursor+1:]
```

Comment:
This can cause `cursor > len(buf)` when the message is not present. It doesn't cause a panic, but it feels wrong to put the cursor in a bad position. So it would be better to check for that:

```go
if len(buf) > *cursor+2 {
	*cursor++
}
*cursor++
```

(these types of edge cases are why long ago compilers started splitting up tokenizing & parsing into two steps :stuck_out_tongue_winking_eye:)